lib/assignDisabledRanges.mjs

```diff
@@ -268,34 +345,58 @@ export default function assignDisabledRanges(root, result) {
 				continue;
 			}
 
-			throw comment.error(`"${ruleToEnable}" has not been disabled`, {
+			throw node.error(`"${ruleToEnable}" has not been disabled`, {
 				plugin: 'stylelint',
 			});
 		}
 	}
 
 	/**
-	 * @param {PostcssComment} comment
+	 * @param {PostcssNode} node
+	 * @param {string} part
+	 * @param {number} offset
+	 */
+	function checkCommentContainingNode(node, part, offset) {
+		tokenize({ css: part })
+			.filter((token) => {
+				return isTokenComment(token);
+			})
+			.forEach((token) => {
+				const [, text, start, end] = token;
+
+				const source = node.rangeBy({
+					index: start + offset,
+					endIndex: end + 1 + offset,
+				});
+
+				checkComment(node, source, text.slice(2, -2).trim());
+			});
+	}
+
+	/**
+	 * @param {PostcssNode} node
+	 * @param {Source | undefined} source
+	 * @param {string} text
 	 */
-	function checkComment(comment) {
+	function checkComment(node, source, text) {
 		// Ignore comments that are not relevant commands
 
-		if (!isConfigurationComment(comment, configurationComment)) {
+		if (!isConfigurationComment(text, configurationComment)) {
 			return;
 		}
 
-		switch (extractConfigurationComment(comment, configurationComment)) {
+		switch (extractConfigurationComment(text, configurationComment)) {
 			case DISABLE_LINE_COMMAND:
-				processDisableLineCommand(comment);
+				processDisableLineCommand(node, source, text);
 				break;
 			case DISABLE_NEXT_LINE_COMMAND:
-				processDisableNextLineCommand(comment);
+				processDisableNextLineCommand(node, source, text);
 				break;
 			case DISABLE_COMMAND:
-				processDisableCommand(comment);
+				processDisableCommand(node, source, text);
 				break;
 			case ENABLE_COMMAND:
-				processEnableCommand(comment);
+				processEnableCommand(node, source, text);
 				break;
```

Comment:
question

Is it OK if the `node` passed to these functions is not a `Comment`?

Reply:
Why would it not be ok?
Can you elaborate, I don't know which cases you are thinking of :)

Reply:
I was just checking if passing `AtRule|Rule|Declaration` to `checkComment` was intended or not.

Reply:
Yes :)

If you read through the changes you will see that it is needed to separate these things:
- the source location
- the comment text
- error reporting

So pass three values to these functions, each dedicated to a specific purpose.